### PR TITLE
Izi modal refactoring - Message fix patch and Popup menu complete

### PR DESCRIPTION
**Pull request by muarachmann**

### CODE REVIEW

**Comment by muarachmann** [comment]
screenshot from 2018-05-03 16-19-48
## Message sreen

screenshot from 2018-05-03 16-20-10
## Add issue screen

File: custom/import_xml.php [hunk 1]
```diff
@@ -185,20 +185,36 @@ function setInsurance($pid, $ainsurance, $asubscriber, $seq) {
...
   else
   {
-  	echo "<html>\n<body>\n<script language='JavaScript'>\n";
-  	//if ($alertmsg)
-	echo " alert('$alertmsg');\n";
-  	/*echo " if (!opener.closed && opener.refreshme) opener.refreshme();\n";
+	  echo "<html>\n<body>\n";
+	  include_once("../interface/globals.php");
+	  include_once("$srcdir/headers.inc.php");
+	  //if ($alertmsg)
```
Comment (tmccormi) [inline custom/import_xml.php]: Don;t leave commented out code.  Just delete it.

File: custom/import_xml.php [hunk 2]
```diff
@@ -185,20 +185,36 @@ function setInsurance($pid, $ainsurance, $asubscriber, $seq) {
...
+	 iziToast.warning({
+               title: 'Warning -',
+               message: '<?php echo($alertmsg); ?>',
+               position: 'bottomRight', // bottomRight, bottomLeft, topRight, topLeft, topCenter, bottomCenter, center
+               icon: 'fa fa-exclamation-triangle'
+
+           });
+
+	";
+	  /*echo " if (!opener.closed && opener.refreshme) opener.refreshme();\n";
```
Comment (tmccormi) [inline custom/import_xml.php]: Same ... delete the commented out code, just makes it hard to review

File: custom/import_xml.php [hunk 3]
```diff
@@ -185,20 +185,36 @@ function setInsurance($pid, $ainsurance, $asubscriber, $seq) {
...
   	echo " window.close();\n";*/
-  	echo "</script>\n</body>\n</html>\n";
-  	//exit();
+	  echo "</script>\n</body>\n</html>\n";
```
Comment (tmccormi) [inline custom/import_xml.php]: don't reformat unnecessarily

File: interface/forms/patient_encounter/common.php [hunk 4]
```diff
@@ -70,12 +70,52 @@ function sensitivity_compare($a, $b) {
...
 
+    $(document).ready(function () {
+        //variables to store titles and subtitles
+        var iziTitle = "";
+        var iziSubTitle = "";
+
+        // New Issue button is clicked.
+        $(".newIssue").click(function () {
+            iziTitle = "<?php echo xlt('Add New Issue'); ?>";
+            iziSubTitle = "<?php echo xlt('Add all sorts of issues about Patient'); ?>";
```
Comment (tmccormi) [inline interface/forms/patient_encounter/common.php]: Hmm, might want to think about the wording of this Subtitle ...

File: interface/forms/patient_encounter/common.php [hunk 5]
```diff
@@ -87,8 +127,14 @@ function saveClicked() {
 
   var category = document.forms[0].pc_catid.value;
   if ( category == '_blank' ) {
-   alert("<?php echo xls('You must select a visit category'); ?>");
-   return;
+   alertMsg = "<?php echo xls('You must select a visit category'); ?>";
+      iziToast.warning({
+          title: 'Warning -',
+          message: alertMsg,
+          position: 'bottomRight', // bottomRight, bottomLeft, topRight, topLeft, topCenter, bottomCenter, center
```
Comment (tmccormi) [inline interface/forms/patient_encounter/common.php]: remove comment here.

File: interface/forms/patient_encounter/save.php [hunk 6]
```diff
@@ -170,6 +189,9 @@
  parent.left_nav.setRadio(window.name, 'enc');
 <?php } // end if new encounter ?>
  parent.left_nav.loadFrame('enc2', window.name, '<?php echo $nexturl; ?>');
+
```
Comment (tmccormi) [inline interface/forms/patient_encounter/save.php]: unnecessary blank lines

File: interface/patient_file/problem_encounter.php [hunk 7]
```diff
@@ -106,12 +107,72 @@
...
 </style>
-
+	<?php call_required_libraries(array("jquery-min-3-1-1","font-awesome" , "iziModalToast")); ?>
 <script type="text/javascript" src="../../library/topdialog.js"></script>
 <script type="text/javascript" src="../../library/dialog.js"></script>
 
+
 <script language="JavaScript">
 
+    $(document).ready(function () {
```
Comment (tmccormi) [inline interface/patient_file/problem_encounter.php]: Is the the same as in common.php above?  If so can we include it instead of copying it?

**Comment by muarachmann** [comment]
okay @tmccormi  seen will review and push back

File: interface/forms/patient_encounter/save.php [hunk 8]
```diff
@@ -146,7 +146,26 @@
...
+        title: 'Success -',
+        message: "Successfully added new encounter",
+        position: 'center',
+        icon: 'fa fa-thumbs-o-up'
+
+    });
+    
+    setTimeout(function () {
+        parent.$("#izi-iframe").iziModal('close');
+    }, 2000);
```
Comment (teryhill) [inline interface/forms/patient_encounter/save.php]: @muarachmann Think about setting a global for the length of time on the messages. Experienced users may want them to disappear faster new installs may need to read them.

File: interface/patient_file/problem_encounter.php (hunk 7 above)
Comment (teryhill) [reply interface/patient_file/problem_encounter.php]: @muarachmann if this code is common to multiple files make a function and put it in a /library/modal_functions.php and just call the function.

**Comment by aethelwulffe** [comment]
Try to give this a lick or two @muarachmann

**Comment by muarachmann** [comment]
sorry @tmccormi , @aethelwulffe have been have serious issues with my internet connection will psuh the commits i have done tonight

**Comment by naveen17797** [comment]
@aethelwulffe yep. Had a lightning strike near my house, which made my modem useless(almost burnt it due to power fluctuation even my home had power safety mechanisms, nature is uncontrollable),for now Mobile Data keeping my internet alive, thats why i push changes to backup things :relaxed:

**Comment by aethelwulffe** [comment]
I live in Tampa Florida, and am from the islands.  I know a little bit 
about lightning... :'(

**Comment by naveen17797** [comment]
Never seen a lightning strike nearer, this was the first time, scary enough.

**Comment by aethelwulffe** [comment]
I hope you folks can get this modal asset hooked up with all the admin/practice stuff as well as Users.  It is REALLY going to help.  Bigtime.
User admin as a modal window may not really be an appropriate approach.  There is a lot of stuff in the user table that is only accessible in the address book, or in the user prefs, and other things.  ACL, facility scheduling and a lot of other stuff would be better served with a spreadsheet-ish sort of admin overview as well.  Managing that stuff one user at a time is quite difficult to track.

**Comment by aethelwulffe** [comment]
Pretty please resolve the conflicts (deleting the conflicting file will work).  This needs testing and review.  Seems complete otherwise.

**Comment by muarachmann** [comment]
@aethelwulffe have a hard time rebasing thoo

**Comment by muarachmann** [comment]
let me see whats up with that custom/import_xml.php file

**Comment by aethelwulffe** [comment]
It was deleted.

**Comment by muarachmann** [comment]
huh

**Comment by muarachmann** [comment]
@tony check the branch now think i resolved the conflicts

**Comment by muarachmann** [comment]
@tmccormi , @aethelwulffe please check that u have the followx screen

**Comment by muarachmann** [comment]
kazam_screenshot_00015

for users and edit users with thier success messages

**Comment by aethelwulffe** [comment]
Hi Mua,
Sorry, testing is taking a long time today.
Say, a while back we suggested making it easier on reviewers by putting in a link to a pr request repo branch.  As we have to pull down a branch to test, there would be no harm in posting something like:
`git fetch https://github.com/muarachmann/libreehr.git iziModalRefactoring:iziModalRefactoring`
in your PR description.  That makes it very easy to copy that line and pull down your branch.

**Comment by aethelwulffe** [comment]
image

**Comment by aethelwulffe** [comment]
Mua,
Do you have a large database for testing?
We have tools to generate large databases, and there is a repo with some pre-made DB inserts.
Some testing is good with a fresh install, but most require some stress testing.
-Example above:  Lots of users shows how bad the calendar layout is...
-Anyway, this retest looks good.  I would adjust the picture thingie sometime in the future so it does not take up a bunch of space for those folks that don't even want to use the feature, but within this scope, this is ready to roll.

**Comment by aethelwulffe** [comment]
Not sure "Add a new user with administrative roles" is a very helpful thing.  It shows up when adding, editing or anything for any user.  I understand that this interface configures actual users, and the same table is used for external contacts via the Address Book, but that is only apparent/meaningful if you have knowledge of the back-end structure.

**Comment by muarachmann** [comment]
Ans still awaiting the invitation

**Comment by aethelwulffe** [comment]
You are a member of the GSOC team Mua, if you still lack permissions, I will bump you more.

Link to a database generator copy (windows exe) here:  http://suncoastconnection.com/downloads/PrintaPatient_A7_2_0.zip

**Comment by aethelwulffe** [comment]
This tests great, and is beautiful.
It is the PR itself that needs cleaning.
Please grab a clean branch repo and move the changes over to it, or otherwise do something to clean this up.  This has been here for a while, and should already be merged.  It is good work, and we need this in the project.
*addressing the delay in globals is probably better not done with yet another global setting that no-one in the field will every change.  Probably best to preserve it as a "never show me this again" or something with user variables, or somewhere to force a change.  I consider this a minor aspect that we can address easily after merging...even if it is immediately after merging.

**Comment by muarachmann** [comment]
@aethelwulffe am doing so henceforth here my commits, making a cleaner branch before evening was hooked up with some shell scripting stuff. As for the link thanks.

**Comment by muarachmann** [comment]
@aethelwulffe closing this PR, it is now here #1207

**Pull request closed without merging**